core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -312,25 +312,30 @@ public int hashCode() {
         return nodeId.hashCode();
     }
 
+    private String toString;
```

Comment:
> I'm not convinced that:

@bleskes Did you take a look at the [place](https://github.com/elastic/elasticsearch/blob/c5bbb023fa8180f0cbc087c3218f3656dbadc7b4/core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java#L609) I mentioned in the `ClusterService` though? This means that the *moment* that a node is added to a cluster, all other nodes in the cluster will have invoked `DiscoveryNode#toString` for the instance representing that node. 

> I would prefer not to calculate the toString all the time. I'm ok with the optimization...

Given this, is preference still to not calculate the toString all the time? I like that the field can be final.